src/core/reference/include/ngraph/runtime/reference/acos.hpp

```diff
@@ -13,7 +13,7 @@ namespace reference {
 template <typename T>
 void acos(const T* arg, T* out, size_t count) {
     for (size_t i = 0; i < count; i++) {
-        out[i] = std::acos(arg[i]);
+        out[i] = static_cast<T>(std::acos(arg[i]));
```

Comment:
should it be acos, acosf and others depending on data type? https://en.cppreference.com/w/cpp/numeric/math/acos